### PR TITLE
Violin plot

### PR DESCRIPTION
**Pull request by JamesxX**
Working on adding violin plots.
- [x] Kernel estimation
- [x] Multiple sets per command
- [x] `both` sides

image

image

### CODE REVIEW

**Comment by JamesxX** [comment]
Further work: Swapping axes. Shouldn't be too difficult, but its not a feature I need for the moment so haven't put the time into implementing it.

File: src/plot/violin.typ [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+///   / both: Plot both sides of the violin.
+/// - kernel (function): The kernel density estimator function.
+/// - bandwidth (float): The smoothing parameter of the kernel.
+/// - extents (float): The extension of the domain, expressed as a fraction of spread.
+/// - samples (int): The number of samples of the kernel to render.
+/// - style (dictionary): Style override dictionary.
+/// - mark-style (dictionary): (unused, will eventually be used to render interquartile ranges).
+/// - axes (axes): (unstable, documentation to follow once completed).
+/// - label (none, content): The name of the category to be shown in the legend.
+#let violin(
```
Comment (johannes-wolf) [inline src/plot/violin.typ:94]: Should be `add-violin`.

File: src/plot/violin.typ [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+///
+/// - data (array): Array of data items. An item is an array containing an `x` and one 
+///                 or more `y` values.
+/// - x-key (int, string): Key to use for retreiving the `x` position of the violin.
+/// - y-key (int, string): Key to use for retreiving values of points within the category.
+/// - side (string): The sides of the violin to be rendered:
+///   / left: Plot only the left side of the violin.
+///   / right: Plot only the right side of the violin.
+///   / both: Plot both sides of the violin.
+/// - kernel (function): The kernel density estimator function.
```
Comment (johannes-wolf) [inline src/plot/violin.typ:86]: Can you add a description of the functions signature here?

File: src/plot/violin.typ (hunk 1 above)
Comment (JamesxX) [reply src/plot/violin.typ]: resolved by 9e0a373

File: src/plot/violin.typ (hunk 2 above)
Comment (JamesxX) [reply src/plot/violin.typ]: resolved by bb654c9

**Review by johannes-wolf — APPROVED** [review]
LGTM.

**Pull request merged**